.github/workflows/build_scan_and_test_rock.yaml

```diff
@@ -0,0 +1,85 @@
+name: Build, scan, and test ROCK
+
+on:
+  workflow_call:
+    inputs:
+      rock-dir:
+        description: "Path to rock directory, i.e. directory containing rockcraft.yaml"
+        required: true
+        default: '.'
+        type: string
+
+jobs:
+  build-scan-test-rocks:
+    name: Build, scan, and test ${{ inputs.rock-dir}}
+    runs-on: ubuntu-22.04
+    steps:
+      # Ideally we'd use self-hosted runners, but this effort is still not stable.
+      # This action will remove unused software (dotnet, haskell, android libs, codeql,
+      # and docker images) from the GH runner, which will liberate around 60 GB of storage
+      # distributed in 40GB for root and around 20 for a mnt point.
+      - name: Maximise GH runner space
+        uses: easimon/maximize-build-space@v7
+        with:
+          root-reserve-mb: 40960
+          remove-dotnet: 'true'
+          remove-haskell: 'true'
+          remove-android: 'true'
+          remove-codeql: 'true'
+          remove-docker-images: 'true'
+
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Build ROCK
+        uses: canonical/craft-actions/rockcraft-pack@main
+        id: rockcraft
+        with:
+          path: ./${{ inputs.rock-dir }}
+
+      - name: Copy ROCK to Docker
+        id: rock_in_docker
+        run: |
+          NAME=$(yq eval ".name" rockcraft.yaml)
+          VERSION=$(yq eval ".version" rockcraft.yaml)
+          ARCH=$(yq eval ".platforms | keys" rockcraft.yaml | awk -F ' ' '{print $2}')
+          ROCK="${NAME}_${VERSION}_${ARCH}"
+          sudo skopeo --insecure-policy copy oci-archive:$ROCK.rock docker-daemon:$ROCK:$VERSION
+          echo "Copied $ROCK:$VERSION to Docker"
+          echo "image=$ROCK:$VERSION" >> "$GITHUB_OUTPUT"
```

Comment:
continuing [this conversation](https://github.com/canonical/seldonio-rocks/pull/64#discussion_r1409833705)

Reply:
Good questions.  Maybe an action is the right thing here?  You can do a composite action (which is essentially just a workflow file, but for some reason...different?)

Reply:
As you mentioned, we can either 
1. make an action
2. leave it as is

I suggest we go with number 2 for now since we won't be copying this over much now that we have reusable workflows. At the same time, creating an action that copies the ROCK could actually be more convoluted and I think this should go in hand with ROCK building issue https://github.com/canonical/charmed-kubeflow-workflows/issues/28. I opened this issue in order to track this https://github.com/canonical/charmed-kubeflow-workflows/issues/29.